src/app/test_executive/send_payment_test.ml

```diff
@@ -0,0 +1,40 @@
+open Async
+open Integration_test_lib
+
+module Make (Engine : Engine_intf) = struct
+  open Engine
+
+  type network = Network.t
+
+  type log_engine = Log_engine.t
+
+  let config =
+    let open Test_config in
+    { default with
+      block_producers= [{balance= "4000"}; {balance= "3000"}]
+    ; num_snark_workers= 0 }
+
+  let run network log_engine =
+    let open Deferred.Or_error.Let_syntax in
+    let block_producer = Caml.List.nth network.Network.block_producers 0 in
+    let%bind () = Log_engine.wait_for_init block_producer log_engine in
+    (* namespace for this particular testnet *)
+    let namespace = network.Network.namespace in
+    (* node in that namespace *)
+    let node = "test-block-producer-1" in
```

Comment:
You are relying on kubernetes naming conventions from inside of the integration test engine. This is a leak of the abstraction. Please pass the `block_producer` node around instead.

Reply:
Passing in an element of the `block_producers` array now.